firmware/application/ui/ui_btngrid.cpp

```diff
@@ -183,16 +192,26 @@ void BtnGridView::insert_item(const GridItem& new_item, size_t position, bool in
 }
 
 void BtnGridView::show_hide_arrows() {
+    portapack::async_tx_enabled = true;
     if (highlighted_item == 0) {
         set_arrow_up_enabled(false);
     } else {
         set_arrow_up_enabled(true);
     }
-    if (highlighted_item == (menu_items.size() - 1)) {
+    if (highlighted_item == (menu_items.size() - 1) || menu_items.size() < displayed_max) {
+        /* ^ when at the last tile of this screen          ^ if the tiles isn't fullfilled this screen */
         set_arrow_down_enabled(false);
     } else {
         set_arrow_down_enabled(true);
     }
+
+    if (menu_items.size() < displayed_max) {  // if the tiles isn't fullfilled this screen
+        button_pgup.hidden(true);
+        button_pgdown.hidden(true);
+    } else {
+        button_pgup.hidden(false);
```

Comment:
Isn't it going to force the display of both arrows whatever the highlighted position is, just because the menu_item.size is bigger than the displayed size ?

Reply:
I think that else is superfluous